podpac/core/cache/cache.py

```diff
@@ -738,39 +540,49 @@ def put(self, node, data, key, coordinates=None, update=False):
         update : bool
             If True existing data in cache will be updated with `data`, If False, error will be thrown if attempting put something into the cache with the same node, key, coordinates of an existing entry.
         '''
-        self.make_cache_dir(node)
-
-        listing = CacheListing(node=node, key=key, coordinates=coordinates, data=data)
-        if self.has(node, key, coordinates): # a little inefficient but will do for now
+        
+        
+        # check for existing entry
+        if self.has(node, key, coordinates):
             if not update:
-                raise CacheException("Existing cache entry. Call put() with `update` argument set to True if you wish to overwrite.")
-            else:
-                paths = self.cache_glob(node, key, coordinates)
-                for p in paths:
-                    c = self.load_container(p)
-                    if c.has(listing):
-                        c.rem(listing)
-                        c.put(listing)
-                        self.save_container(c,p)
-                        return True
-                raise CacheException("Data is cached, but unable to find for update.")
-        # listing does not exist in cache
-        path = self.cache_path(node, key, coordinates)
-        # if file for listing already exists, listing needs to be added to file
-        if self.file_exists(path):
-            c = self.load_container(path)
-            c.put(listing)
-            self.save_container(c,path)
-        # if file for listing does not already exist, we need to create a new container, add the listing, and save to file
+                raise CacheException("Existing cache entry. Use `update=True` to overwrite.")
+            self.rem(node, key, coordinates)
+
+        # serialize
+        rootpath = self.cache_path(node, key, coordinates)
+        
+        if isinstance(data, xr.DataArray):
+            path = rootpath + '.netcdf'
+            s = data.to_netcdf()
+        elif isinstance(data, np.ndarray):
+            path = rootpath + '.npy'
+            with io.BytesIO() as f:
+                np.save(f, data)
+                s = f.getvalue()
+        elif isinstance(data, podpac.Coordinates):
```

Comment:
you could also add an entry for nodes, which also have the JSON property...

Reply:
I left it out originally because it seems like a weird thing to cache, but I pushed this. The implementation is weird because we don't have a `from_json` for nodes in general, we only have it for `Pipeline` nodes.

Reply:
Well, I certainly didn't think that through... well, now that it's in I think it's fine to leave. We'll likely fin d a use for it sometime.